src/lib/coda_version/BUILD.bazel

```diff
@@ -0,0 +1,78 @@
+load( "@obazl_rules_ocaml//ocaml:rules.bzl", "ocaml_interface", "ocaml_module")
+load( "@obazl_tools_bazel//tools:rules.bzl", "stamped_filegroup")
+
+## Only stamp if --stamp flag is pased.  Otherwise use generic defaults.
+
+#############
+ocaml_module(
+    name = "coda_version",
+    module_name = "coda_version",
+    src  = select({
+        "//:stampit": "coda_version.ml",
+        "//conditions:default": "coda_version_defaults.ml"
+    }),
+    opts = [
+        "-w", "-32" # Error (warning 32): unused value marlin_branch.
+    ],
+    intf = ":_Coda_version.cmi",
+    ppx = "//bzl/ppx/exe:ppx_version",
+    visibility = ["//visibility:public"],
+)
+
+################
+ocaml_interface(
+    name = "_Coda_version.cmi",
+    src = "coda_version.mli",
+    ppx = "//bzl/ppx/exe:ppx_version"
+)
+
+########
+genrule(
+    name = "gensrc",
+    stamp = True,
+    srcs = ["coda_version.ml.template"],
+    outs = ["coda_version.ml"],
+    cmd = "\n".join([
```

Comment:
Is it possible to use environment variables somehow instead of an inline bash script?

Reply:
I don't think so.  This is designed to work with Bazel stamping.  This code is provisional; it works for now, but its a change from the current approach so you folks will need to learn how Bazel stamping works and decide if this is the right way to go.

I've opened an issue on this, https://github.com/MinaProtocol/mina/issues/7163

Reply:
Great, thanks